File: app/src/main/java/com/mrboomdev/awery/catalog/template/CatalogTag.java
package com.mrboomdev.awery.catalog.template;

public class CatalogTag {
	public String name, id, description;
	public boolean isAdult, isSpoiler;

	public CatalogTag(String name, String description, boolean isAdult) {
		this.name = name;
		this.description = description;
		this.isAdult = isAdult;
	}

	public CatalogTag() {}

	public String getName() {
		return name;
	}

	public String getDescription() {
		return description;
	}

	public boolean isAdult() {
		return isAdult;
	}
}

File: ext/src/main/java/com/mrboomdev/awery/ext/data/CatalogTag.kt
package com.mrboomdev.awery.ext.data

import com.squareup.moshi.JsonClass
import java.io.Serial
import java.io.Serializable

@JsonClass(generateAdapter = true)
@kotlinx.serialization.Serializable
class CatalogTag @JvmOverloads constructor(
	val name: String,
	val description: String? = null,
	val isAdult: Boolean = false,
	val isSpoiler: Boolean = false
) : Serializable {
	companion object {
		@Serial
		private val serialVersionUID: Long = 1
	}
}